packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1598,6 +1612,81 @@ function updateSuspenseComponent(
   return next;
 }
 
+function updateDehydratedSuspenseComponent(
+  current: Fiber | null,
+  workInProgress: Fiber,
+  renderExpirationTime: ExpirationTime,
+) {
+  if (current === null) {
+    // During the first pass, we'll bail out and not drill into the children.
+    // Instead, we'll leave the content in place and try to hydrate it later.
+    workInProgress.expirationTime = Never;
+    return null;
+  }
+  // We use childExpirationTime to indicate that a child might depend on context, so if
+  // any context has changed, we need to treat is as if the input might have changed.
+  const hasContextChanged = current.childExpirationTime >= renderExpirationTime;
+
+  const prevProps = current.memoizedProps;
+  const nextProps = workInProgress.pendingProps;
+
+  if (hasContextChanged || prevProps !== nextProps) {
```

Comment:
We already compare `prevProps` to `nextProps` at the beginning of `beginWork`. And if they are different we set the module-level boolean `didReceiveUpdate` to true. (This approach would also account for legacy context changing, even though that's deprecated in strict mode anyway and I'm guessing partial hydration will only work in strict mode?)